src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -294,6 +294,71 @@ public Mnemonic CreateWallet(string password, string name, string passphrase, Mn
             return mnemonic;
         }
 
+        /// <inheritdoc />
+        public string SignMessage(string password, string walletName, string externalAddress, string message)
+        {
+            Guard.NotEmpty(password, nameof(password));
+            Guard.NotEmpty(walletName, nameof(walletName));
+            Guard.NotEmpty(message, nameof(message));
+            Guard.NotEmpty(externalAddress, nameof(externalAddress));
+
+            string signature = string.Empty;
+
+            // Get wallet
+            Wallet wallet = this.GetWalletByName(walletName);
+
+            // Check the password.
+            try
+            {
+                if (!wallet.IsExtPubKeyWallet)
+                    Key.Parse(wallet.EncryptedSeed, password, wallet.Network);
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogTrace("Exception occurred: {0}", ex.ToString());
+                this.logger.LogTrace("(-)[EXCEPTION]");
+                throw new SecurityException(ex.Message);
+            }
+
+            // Sign the message.
+            try
+            {
+                HdAddress hdAddress = wallet.FindHDAddressByExternalAddress(externalAddress);
+                Key privateKey = wallet.GetExtendedPrivateKeyForAddress(password, hdAddress).PrivateKey;
+                BitcoinSecret secret = this.network.CreateBitcoinSecret(privateKey);
```

Comment:
Are you sure you need this? Can't you just do `privateKey.SignMessage(message);`?

Reply:
Ahhh, Great catch, good stuff, thanks!